lib/gobl/bill/invoice.rb

```diff
@@ -29,10 +29,28 @@ class Invoice < GOBL::Object
       property :code, String
       validates :code, presence: true
 
+      # Inline enum type for the `type` property
+      class TypeEnum < GOBL::CBC::Key
```

Comment:
Okay, I get what you mean. This feels a bit strange. How important do you think this is to have in Ruby? Perhaps it's easier just to define a list and use the `validates` method against it as opposed to defining a dictionary?

Reply:
How about using the validates like this:
```ruby
class Invoice < GOBL::Object
  TYPES = {
    "standard" => "Standard description...",
    "proforma" => "Proforma description..."
  }
  property :type, GOBL::CBC::Key
  validates :type, inclusion: { in: TYPES.keys }
```

Reply:
That's exactly what I implemented at first! But in doing so, I lost some of the niceties that Enum provided:

1. The ability to compare with strings `invoice.type == 'credit-note'`. 
2. ...or with symbols `invoice.type == :credit_note`.
3. ...or using enquiry methods `invoice.type.credit_note?`.
4. ...or get the description directly `invoice.type.description #=> "Credit note description"`.

And that's why I implemented the inline enum solution.

Now, I spent some time yesterday applying the changes to invopop/pdf, invopop/plemsi and invopop/invoicexpress and realised that we hardly use these features and that they don't add much value. Also, at the end of the day, these fields are JSON strings, and there is a case for explicitness over "magic".

So I decided to prepare an alternative PR that simplifies `gobl.ruby` a lot by removing most of the complexity. It leaves the 20% of it that provides the 80% of the value (IMO). That is:

1. The ability to validate the inclusion of restricted JSON attributes to the allowed values.
2. The ability to compare and treat value objects as if they were base class objects (i.e. usually strings). That is, being able to do `invoice.type == 'credit-note'` instead of `invoice.type.to_s == 'credit-note'`.

Reply:
Could you not just extend the `GOBL::CBC::Key` class with a custom `def ==(val)` method? In Go we like using constants for everything anyway, as they give you automatic type safety for comparisons :-)

Proposal sounds good though, I'm in favour of simplicity!

Reply:
Yep! That's what `GOBL::Value` and `GOBL::Enum` values did. There's also other comparison-related methods such as `===`, `eql?` or `hash` that were also being overridden. But that only solved comparison. What about doing, for instance, something like `tax_id.code[0..6]`? You have no option but doing `tax_id.code.to_s[0..6]` which is not a big deal but in plain JSON that value is just a String. So, I thought perhaps the right™️ way of solving this was making class for non-object structs, like `GOBL::CBC::Key` or `GOBL::CBC::Code`, to inherit directly from the base type defined in the schema (usually `String`). I gave a try, and seems to work pretty well. PRs coming soon!